### DISCUSSION TITLE
Esc-key keeps re-opening panels.

### DISCUSSION THREAD

**Discussion by JuriTuuti** [phcode-dev/phoenix#2001] [General]
Whichever panel was last open and that I closed -- key shortcuts, html errors.... keeps re-opening every time I press Esc for some other reason (like when I want to close typing suggestions).  The Esc key seems to have an additional function now - in addition to doing what it is supposed to do in the editor, it also toggles the bottom panel on and off.

If I now want to escape a text suggestion, I need to hit Esc 2 times.  Once to close the suggestion and open bottom panel, and a second time to close the bottom panel.

Then a general suggestion.  Put brakes on all the new features you are planning, and make the basic functions of the editor work flawlessly first.   Right now, it seems I spend more time trying to work around the problems than with what I am supposed to be doing - so much so I just re-installed Brackets.

**Comment by abose** [comment]
Thanks for reporting the issue and this looks like a serious workflow breaker bug that we missed in the QA. pressing escape key to toggle the last close panel was a feature we added, but it should not have activated when code hints are opened. Looks like this happens when code hints are activated without pressing `ctrl-space` which is the majority of the cases. This will be fixed in the next release on Jan.

**Reply by abose** [reply]
fixed by https://github.com/phcode-dev/phoenix/pull/2002
